contracts/EtomicSwapNft.sol

```diff
@@ -0,0 +1,719 @@
+        PaymentSent,
+        TakerApproved,
+        MakerSpent,
+        TakerRefunded
+    }
+
+    struct TakerPayment {
+        bytes20 paymentHash;
+        uint32 preApproveLockTime;
+        uint32 paymentLockTime;
+        TakerPaymentState state;
+    }
+
+    event TakerPaymentSent(bytes32 id);
+    event TakerPaymentApproved(bytes32 id);
+    event TakerPaymentSpent(bytes32 id, bytes32 secret);
+    event TakerPaymentRefundedSecret(bytes32 id, bytes32 secret);
+    event TakerPaymentRefundedTimelock(bytes32 id);
+
+    mapping(bytes32 => TakerPayment) public takerPayments;
+
+    address public immutable dexFeeAddress;
+
+    constructor(address feeAddress) {
+        require(
+            feeAddress != address(0),
+            "feeAddress must not be zero address"
+        );
+
+        dexFeeAddress = feeAddress;
+    }
+
+    function spendErc721MakerPayment(
+        bytes32 id,
+        address maker,
+        bytes32 takerSecretHash,
+        bytes32 makerSecret,
+        address tokenAddress,
+        uint256 tokenId
+    ) external {
+        require(
+            makerPayments[id].state == MakerPaymentState.PaymentSent,
+            "Invalid payment state. Must be PaymentSent"
+        );
+
+        // Check if the function caller is an externally owned account (EOA)
+        require(msg.sender == tx.origin, "Caller must be an EOA");
+
+        bytes20 paymentHash = ripemd160(
+            abi.encodePacked(
+                msg.sender,
+                maker,
+                takerSecretHash,
+                sha256(abi.encodePacked(makerSecret)),
+                tokenAddress,
+                tokenId
+            )
+        );
+        require(
+            paymentHash == makerPayments[id].paymentHash,
+            "Invalid paymentHash"
+        );
+
+        // Effects
+        makerPayments[id].state = MakerPaymentState.TakerSpent;
+
+        // Event Emission
+        emit MakerPaymentSpent(id);
+
+        // Interactions
+        IERC721 token = IERC721(tokenAddress);
+        token.safeTransferFrom(address(this), msg.sender, tokenId);
+    }
+
+    function spendErc1155MakerPayment(
+        bytes32 id,
+        uint256 amount,
+        address maker,
+        bytes32 takerSecretHash,
+        bytes32 makerSecret,
+        address tokenAddress,
+        uint256 tokenId
+    ) external {
+        require(
+            makerPayments[id].state == MakerPaymentState.PaymentSent,
+            "Invalid payment state. Must be PaymentSent"
+        );
+
+        // Check if the function caller is an externally owned account (EOA)
+        require(msg.sender == tx.origin, "Caller must be an EOA");
+
+        bytes20 paymentHash = ripemd160(
+            abi.encodePacked(
+                msg.sender,
+                maker,
+                takerSecretHash,
+                sha256(abi.encodePacked(makerSecret)),
+                tokenAddress,
+                tokenId,
+                amount
+            )
+        );
```

Comment:
Shouldn't we put amount at the start similar to ETH/ERC20 swap  contract https://github.com/KomodoPlatform/etomic-swap/blob/124445a6f1cd61892bdcaa3cec786eee91b7e5f5/contracts/EtomicSwapV2.sol#L131-L140

Reply:
From my point of view, when it is about NFT, the fields which describe this feature are: tokenAddress, tokenId, amount (erc1155). Thats why I prefer to stick amount with tokenAddess, tokenId together.